src/middleware/authentication.js

```diff
@@ -0,0 +1,22 @@
+const jwt = require('jsonwebtoken');
+const { UnauthenticatedError } = require('../errors');
+
+const auth = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader || !authHeader.startsWith('Bearer ')) {
+    throw new UnauthenticatedError('Authentication invalid');
+  }
+
+  const token = authHeader.split(' ')[1];
+
+  try {
```

Comment:
Lets make sure we check for process.env.JWT_SECRET before using it or it could crash.